src/sentry/db/router.py

```diff
@@ -0,0 +1,70 @@
+import django.apps
+
+DB_TABLE_TO_DB = {}
+
+
+def db_for_model(model_class):
+    if model_class._meta.app_label == "sentry":
+        return list(model_class._meta._ModelSiloLimit__silo_limit.modes)[0]._name_.lower() + "_silo"
+    return "default"
+
+
+def db_for_table(table):
+    # TODO: cache this
+    # Fill DB_TABLE_TO_DB with sentry models
+    for model_class in django.apps.apps.get_models():
+        if model_class._meta.app_label == "sentry":
+            DB_TABLE_TO_DB[model_class._meta.db_table] = (
+                list(model_class._meta._ModelSiloLimit__silo_limit.modes)[0]._name_.lower()
+                + "_silo"
+            )
+    return DB_TABLE_TO_DB.get(table, "default")
```

Comment:
Perhaps this mapping is something we can generate during application startup via the decorators?

Reply:
I attempted that first. At initialization time, `django.apps.apps.get_models()` hard fails with an exception because the models aren't registered yet, so either it needs to be loaded later (something like the above) or we need to somehow change the initialization order.